packages/landing/src/map.ts

```diff
@@ -0,0 +1,107 @@
+import View from 'ol/View';
+import TileLayer from 'ol/layer/Tile';
+import XYZ from 'ol/source/XYZ';
+import * as proj from 'ol/proj.js';
+import WMTS from 'ol/source/WMTS.js';
+import { Epsg } from '@basemaps/geo';
+import { NztmOl } from './nztm2000';
+import Layer from 'ol/layer/Layer';
+
+/** Projection to use for the URL bar */
+const UrlProjection = 'EPSG:4326';
+
+/** Default center point if none provided */
+const DefaultCenter: Record<number, MapLocation> = {
+    [Epsg.Google.code]: { lat: -41.277848, lon: 174.7763921, zoom: 8 },
+    [Epsg.Nztm2000.code]: { lat: -41.277848, lon: 174.6763921, zoom: 2 },
+};
+
+const attributions = '<a href="http://data.linz.govt.nz">LINZ CC BY 4.0 © Imagery Basemap contributors</a>';
+
+export class Basemaps {
+    map: Map;
+    el: HTMLElement;
+    config: MapOptions;
+    location: Location;
+
+    constructor(target: HTMLElement) {
+        this.el = target;
+        this.updateFromUrl();
+    }
+
+    getLayer(): Layer {
+        const projection = this.config.projection;
+
+        if (projection == Epsg.Google) {
+            return new TileLayer({
+                source: new XYZ({ url: WindowUrl.toTileUrl(this.config, MapOptionType.Tile), attributions }),
+            });
+        }
+
+        if (projection == Epsg.Nztm2000) {
+            return new TileLayer({
+                source: new WMTS({
+                    url: WindowUrl.toTileUrl(this.config, MapOptionType.TileWmts),
+                    requestEncoding: 'REST',
+                    attributions: attributions,
+                    projection: projection.toEpsgString(),
+                    tileGrid: NztmOl.TileGrid,
+                    // These keys arent really needed but need to be strings
+                    layer: '',
+                    style: '',
+                    matrixSet: '',
+                }),
+            });
+        }
+
+        throw new Error('Unable to find layer for projection: ' + projection);
+    }
+    /** Import location and rendering information from the url */
+    private updateFromUrl(): void {
+        this.config = WindowUrl.fromUrl(window.location.search);
+        const projection = this.config.projection;
+        const location = { ...DefaultCenter[projection.code], ...WindowUrl.fromHash(window.location.hash) };
+
+        const loc = proj.transform([location.lon, location.lat], UrlProjection, `EPSG:${projection}`);
+        let resolutions: undefined | number[] = undefined;
+        if (projection == Epsg.Nztm2000) {
+            resolutions = NztmOl.resolutions;
+        }
+        const view = new View({ projection: projection.toEpsgString(), center: loc, zoom: location.zoom, resolutions });
+
+        this.map = new Map({
+            target: this.el,
+            view,
+            layers: [this.getLayer()],
+        });
+
+        this.map.addEventListener('postrender', this.postRender);
+    }
+
+    updateUrlTimer: unknown | null = null;
```

Comment:
It would be better for this to be zero when no timer is running rather than switching data types between null and number

Reply:
2 problems.

1. null vs 0, 0 would be a convention, to know that 0 is not a valid timer number (is it?) I would prefer to be explict and say that if its null then there is none.

2. there is a weird typing issue with timers in DOM vs Node, since we have both sets of types imported, hence why its set as a unknown.